Regression: system_health.test_system_health test_service_checker_with_process_exit failing after addition of check_system_health_led_info

### Is it platform specific

generic

### Importance or Severity

Critical

### Previous Working Version

Prior to #20716 merging

### Steps to Reproduce

Run `system_health/test_system_health::test_service_checker_with_process_exit`, and observe the following failure:
```

        if all(status == "OK" for status in status_dict.values()):
            assert system_led_status.lower() == 'green', \
                f"System status LED is not green, but it is {system_led_status}"
        else:
>           assert system_led_status.lower() in ["yellow", "amber", "red"], \
                f"System status LED is not yellow, amber, or red, but it is {system_led_status}"
E           AssertionError: System status LED is not yellow, amber, or red, but it is green

duthost    = <MultiAsicSonicHost gold228>
status_data = [('Services', 'Not'), ('Hardware', 'OK')]
status_dict = {'Hardware': 'OK', 'Services': 'Not'}
system_health_summary = 'Invalid color\nSystem status summary\n\n  System status LED  green\n  Services:\n    Status: Not OK\n    Not Running: snmp:snmpd\n  Hardware:\n    Status: OK'
system_led_res = ['green']
system_led_status = 'green'

system_health/test_system_health.py:520: AssertionError
```
At this point in the test, a critical process has been shut down by the test to ensure that the system health status has changed to unhealthy. This will attempt to set the system LED to the `fault` status. We then call `check_system_health_led_info` which checks that the system status LED is yellow, amber, or red:
`assert system_led_status.lower() in ["yellow", "amber", "red"]`
However, as part of the setup for this test, `ConfigFileContext(duthost, os.path.join(FILES_DIR, IGNORE_DEVICE_CHECK_CONFIG_FILE))` has been invoked which puts https://github.com/sonic-net/sonic-mgmt/blob/master/tests/system_health/files/ignore_device_check.json in place of the system's default health checker config. This contains:
```
    "led_color": {
      "fault": "orange",
```

This creates 2 possibilties for how this test can go:
1. If the device doesn't support "orange" as a system LED color, it will fail to set the color and fail the test because the LED is still green/healthy. We see this in the test above, note the `Invalid color` log before the normal output in `system_health_summary`.
2. If the device does support "orange", the device will set its system led to orange/fault and the test will fail because the color isn't in the checked set, yellow, amber, and red.

Reviewing the test, I'm not sure it's possible to pass as is.

We can fix this for case 1 above by omitting the `ConfigFileContext` for this test, I'm not sure it is required for this test. I'll send a PR with this fix.

For number 2, we could add orange to the list of "bad" colors in check_system_health_led_info.

However, I think there is a more fundamental problem with the test_system_health.py tests overriding the system health checker config with colors that the devices don't support. I'm not sure what the best way to solve that is.

### Impact of this regression

The test fails when it shouldn't.

### Relevant log output

```shell

```

### Output of `show version`

```shell

```

### Attach files (if any)

_No response_

--- a/tests/system_health/test_system_health.py
+++ b/tests/system_health/test_system_health.py
@@ -542,13 +542,26 @@ class ConfigFileContext:
 
     def __enter__(self):
         """
-        Back up original system health config file and replace it with the given one.
+        Back up original system health config and merge updates from source file.
+        This should preserve platform-specific settings like led_color.
         :return:
         """
         self.dut.command(
             'mv -f {} {}'.format(self.origin_config, self.backup_config))
-        self.dut.copy(src=os.path.join(FILES_DIR, self.src),
-                      dest=self.origin_config)
+
+        output = self.dut.command('cat {}'.format(self.backup_config))
+        origin = json.loads(output['stdout'])
+
+        with open(self.src, 'r') as f:
+            updates = json.load(f)
+
+        # Preserve platform-specific settings if it exists in the original config
+        if 'led_color' in origin:
+            updates.pop('led_color', None)
+
+        origin.update(updates)
+
+        self.dut.copy(content=json.dumps(origin, indent=4), dest=self.origin_config)
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         """

